Tests/SonarScanner.MSBuild.PreProcessor.Test/SonarQubeWebServiceTest.cs

```diff
@@ -386,16 +390,37 @@ public async Task GetProperties_RequestUrl(string hostUrl, string version, strin
         [DataRow("http://myhost:222/sonar/", "http://myhost:222/sonar/api/editions/is_valid_license")]
         public async Task IsServerLicenseValid_RequestUrl(string hostUrl, string licenseUrl)
         {
-            var testDownloader = new TestDownloader();
-            testDownloader.Pages[new Uri(licenseUrl)] = @"{ ""isValidLicense"": true }";
-            sut = new SonarQubeWebService(testDownloader, new Uri(hostUrl), version, logger);
+            downloader.Pages[new Uri(licenseUrl)] = @"{ ""isValidLicense"": true }";
+            sut = new SonarQubeWebService(downloader, new Uri(hostUrl), version, logger);
 
             var isValid = await sut.IsServerLicenseValid();
 
             isValid.Should().BeTrue();
         }
 
         [TestMethod]
+        public async Task DownloadCache_NullArgument()
+        {
+            (await sut.Invoking(x => x.DownloadCache(null)).Should().ThrowAsync<ArgumentNullException>()).And.ParamName.Should().Be("localSettings");
+        }
+
+        [TestMethod]
+        [DataRow("9.8", "", "", "Incremental PR analysis is available starting with SonarQube 9.9 or later.")]
+        [DataRow("9.9", "", "", "ProjectKey parameter was not provided. Incremental PR analysis is disabled.")]
+        [DataRow("9.9", "BestProject", "", "Base branch parameter was not provided. Incremental PR analysis is disabled.")]
+        public async Task DownloadCache_InvalidArguments(string version, string projectKey, string branch, string debugMessage)
+        {
+            sut = new SonarQubeWebService(downloader, serverUrl, new Version(version), logger);
+            var localSettings = GetLocalSettings(projectKey, branch);
+
+            var res = await sut.DownloadCache(localSettings);
```

Comment:
We should rename `res` to `cache`, `cacheEntries` or `result` otherwise it's not following our guidelines (no abbreviations).